scripts/rescript_bsb.js

```diff
@@ -6,47 +6,47 @@ var os = require("os");
 const child_process = require("child_process");
 const rescript_exe = require("./bin_path").rescript_exe;
 
-/**
- * @typedef {Object} ProjectFiles
- * @property {Array<string>} dirs
- * @property {Array<string>} generated
- */
-
-/**
- * @typedef {Object} WatcherRef
- * @property {string} dir
- * @property {fs.FSWatcher} watcher
- */
-
 const cwd = process.cwd();
 const lockFileName = path.join(cwd, ".bsb.lock");
 
-let isBuilding = false;
+/**
+ * @type {child_process.ChildProcess | null}
+ */
+let ownerProcess = null;
 function releaseBuild() {
-  if (isBuilding) {
+  if (ownerProcess) {
     try {
+      ownerProcess.kill();
       fs.unlinkSync(lockFileName);
-    } catch (err) {}
-    isBuilding = false;
+    } catch {}
+    ownerProcess = null;
   }
 }
 
-// We use [~perm:0o664] rather than our usual default perms, [0o666], because
-// lock files shouldn't rely on the umask to disallow tampering by other.
-function acquireBuild() {
-  if (isBuilding) {
-    return false;
+/**
+ * We use [~perm:0o664] rather than our usual default perms, [0o666], because
+ * lock files shouldn't rely on the umask to disallow tampering by other.
+ *
+ * @param {Array<string>} args
+ */
+function acquireBuild(args) {
+  if (ownerProcess) {
+    return null;
   } else {
     try {
-      const fid = fs.openSync(lockFileName, "wx", 0o664);
-      fs.closeSync(fid);
-      isBuilding = true;
+      ownerProcess = child_process.spawn(rescript_exe, args, {
+        stdio: "inherit",
+      });
```

Comment:
Should it be here?

Reply:
I've moved it to manage the spawned process itself with the lockfile.

To prevent the dangling process, a kill signal is explicitly sent upon release, and even in case of failure, the pid is left in the lockfile so that the user can control it directly.

Reply:
It'll trigger the build twice for watch mode